nix/pkgs.nix

```diff
@@ -3,10 +3,22 @@
 let
   sources = import ./sources.nix {};
   pkgs1903 = import sources."nixpkgs-19.03" {};
-  cardanoNodePkgs = import sources.cardano-node { inherit system crossSystem config; };
+  commonLib' = import ./default.nix {};
+  commonLib = commonLib'.commonLib;
 in pkgs: super: with pkgs; {
   jmPkgs = import ./jormungandr.nix { inherit (pkgs) commonLib; inherit pkgs; };
+  cardanoNodePkgs = import sources.cardano-node { inherit system crossSystem config; };
   cardano-node = cardanoNodePkgs.cardano-node // {
+    mainnet = {
+      configFile =
+        let
+          rawCfg = commonLib.cardanoLib.environments.mainnet.nodeConfig // {
```

Comment:
Would be much neater and less risky for typos to not replicate `mainnet` three times, but didn't try figuring out how to do that. 

Saw that it was [possible](https://github.com/input-output-hk/cardano-node/blob/f9d541c51720811897310d09df8b8e43deae9570/nix/scripts.nix#L90) though.

Reply:
Sometimes a little repetition is ok if the factorized version is complicated.

This seems ok anyway. I had to do a similar thing in [cardano-launcher/test/utils.ts](https://github.com/input-output-hk/cardano-launcher/blob/1bfb55114d7d5c4fe0f9bd3454d06cdc9f6f0a43/test/utils.ts#L125-L173) (except that I edited the config file with javascript rather than nix).

Reply:
> Sometimes a little repetition is

I know. But here I suspect the factorised version should be clearly better. I was very close to returning the testnet config for this mainnet helper by accident.